api/pacman-api-admin/src/main/java/com/tmobile/pacman/api/admin/repository/service/RedHatAccountServiceImpl.java

```diff
@@ -0,0 +1,239 @@
+            // Make the API request
+            HttpResponse response = httpClient.execute(request);
+            // Get the HTTP status code
+            int statusCode = response.getStatusLine().getStatusCode();
+
+            // Create an AccountValidationResponse based on the status code
+            if (statusCode == 200) {
+                accountResponse.setValidationStatus(SUCCESS);
+                accountResponse.setMessage(VALIDATION_SUCCESSFUL);
+                LOGGER.info(VALIDATION_SUCCESSFUL);
+            } else {
+                accountResponse.setValidationStatus(FAILURE);
+                accountResponse.setMessage(TOKEN_VALIDATION_FAILED);
+                accountResponse.setErrorDetails(String.format(ERROR_MESSAGE_TEMPLATE +
+                        HTTP_STATUS_CODE_TEMPLATE, VALIDATING, statusCode));
+                LOGGER.info(VALIDATION_FAILED);
+                LOGGER.info(String.format(ERROR_MESSAGE_TEMPLATE +
+                        HTTP_STATUS_CODE_TEMPLATE, VALIDATING, statusCode));
+            }
+        } catch (UnknownHostException e) {
+            accountResponse.setValidationStatus(FAILURE);
+            accountResponse.setMessage(ACCOUNT_ID_VALIDATION_FAILED);
+            accountResponse.setErrorDetails(String.format(ERROR_MESSAGE_TEMPLATE + e.getMessage(), VALIDATING));
+            LOGGER.info(ACCOUNT_ID_VALIDATION_FAILED);
+            LOGGER.info(String.format(ERROR_MESSAGE_TEMPLATE + e.getMessage(), VALIDATING));
+        } catch (Exception e) {
+            accountResponse.setValidationStatus(FAILURE);
+            accountResponse.setMessage(VALIDATION_FAILED);
+            accountResponse.setErrorDetails(String.format(ERROR_MESSAGE_TEMPLATE + e.getMessage(), VALIDATING));
+            LOGGER.info(VALIDATION_FAILED);
+            LOGGER.info(String.format(ERROR_MESSAGE_TEMPLATE + e.getMessage(), VALIDATING));
+        }
+        return accountResponse;
+    }
+
+    @Override
+    public AccountValidationResponse addAccount(CreateAccountRequest accountData) {
+        LOGGER.info(VALIDATING_REDHAT_DETAILS);
+        if (accountData.getRedhatAccountName() == null || accountData.getRedhatAccountName().isEmpty()) {
+            accountData.setRedhatAccountName(accountData.getRedhatAccountId());
+        }
+        AccountValidationResponse accountResponse = validateRequest(accountData);
+        if (accountResponse.getValidationStatus().equalsIgnoreCase(FAILURE)) {
+            LOGGER.info(VALIDATION_FAILED_DUE_TO_MISSING_PARAMETERS);
+            return accountResponse;
+        }
+        LOGGER.info(ADDING_ACCOUNT);
+        AccountValidationResponse validateResponse;
+        try {
+            validateResponse = createAccountInDb(accountData.getRedhatAccountId(), accountData.getRedhatAccountName(),
+                    Constants.REDHAT, accountData.getCreatedBy());
+
+            if (validateResponse.getValidationStatus().equalsIgnoreCase(FAILURE)) {
+                LOGGER.info(ACCOUNT_EXISTS);
```

Comment:
How do you know that the DB failure is because Account already exists? what if DB is offline?

Reply:
![image](https://github.com/PaladinCloud/CE/assets/114460784/a3bdcbfc-80b4-45f5-af3b-8c6bdaea16fc)
I reused the same (createAccountInDb) function in the AbstractAccountServiceImpl, anyway if we fail we need to stop the process